CRM/Sepa/Logic/Queue/Update.php

```diff
@@ -23,121 +23,182 @@
+        // Safety margin
+        $count = self::getMandateCount($creditor['id'], $sdd_mode) + SDD_UPDATE_RUNNER_BATCH_SIZE;
+
+        for ($offset=0; $offset < $count; $offset += SDD_UPDATE_RUNNER_BATCH_SIZE) {
+          $queue->createItem(self::createTask('UPDATE', [
+            'creditor_id' => $creditor['id'],
+            'limit'       => SDD_UPDATE_RUNNER_BATCH_SIZE,
+            'mode'        => $sdd_mode,
+            'offset'      => $offset,
+          ]));
         }
-        $queue->createItem(new CRM_Sepa_Logic_Queue_Update('CLEANUP', $sdd_mode));
+
+        $queue->createItem(self::createTask('CLEANUP', [ 'mode' => $sdd_mode ]));
       }
     }
 
-    $queue->createItem(new CRM_Sepa_Logic_Queue_Update('FINISH', $mode));
-
-    // create a runner and launch it
-    $runner = new CRM_Queue_Runner(array(
-      'title'     => ts("Updating %1 SEPA Groups", array(1 => $mode, 'domain' => 'org.project60.sepa')),
-      'queue'     => $queue,
-      'errorMode' => CRM_Queue_Runner::ERROR_ABORT,
-      // 'onEnd'     => array('CRM_Admin_Page_ExtensionsUpgrade', 'onEnd'),
-      'onEndUrl'  => CRM_Utils_System::url('civicrm/sepa/dashboard', 'status=active'),
-    ));
-    $runner->runAllViaWeb(); // does not return
-  }
-
-
-  protected function __construct($cmd, $mode, $creditor_id = NULL, $offset = NULL, $limit = NULL) {
-    $this->cmd         = $cmd;
-    $this->mode        = $mode;
-    $this->creditor_id = $creditor_id;
-    $this->offset      = $offset;
-    $this->limit       = $limit;
+    $queue->createItem(self::createTask('FINISH'));
 
-    // set title
-    switch ($this->cmd) {
-      case 'PREPARE':
-        $this->title = ts("Preparing to clean up ended mandates", array('domain' => 'org.project60.sepa'));
-        break;
+    $bgqueue_enabled = (bool) Civi::settings()->get('enableBackgroundQueue');
 
-      case 'CLOSE':
-        $this->title = ts("Cleaning up ended mandates", array('domain' => 'org.project60.sepa'));
-        break;
+    if (!$bgqueue_enabled) {
+      $runner_title = ts('Updating %1 SEPA Groups', [
+        1        => $mode,
+        'domain' => 'org.project60.sepa',
+      ]);
 
-      case 'UPDATE':
-        $this->title = ts("Process {$this->mode} mandates (%1-%2)",
-          array(1 => $this->offset, 2 => $this->offset+$this->limit, 'domain' => 'org.project60.sepa'));
-        break;
+      $runner = new CRM_Queue_Runner([
+        'errorMode' => CRM_Queue_Runner::ERROR_ABORT,
+        'onEndUrl'  => CRM_Utils_System::url('civicrm/sepa/dashboard', 'status=active'),
+        'queue'     => $queue,
+        'title'     => $runner_title,
+      ]);
 
-      case 'CLEANUP':
-        $this->title = ts("Cleaning up {$this->mode} groups",
-          array(1 => $this->offset, 2 => $this->offset+$this->limit, 'domain' => 'org.project60.sepa'));
-        break;
+      $runner->runAllViaWeb();
```

Comment:
Documentation mentions a `runAllInteractive()` method which should automatically respect `enableBackgroundQueue`. Could we use that?